timely_beliefs/sensors/func_store/knowledge_horizons.py

```diff
@@ -31,6 +34,45 @@ def at_date(
     return event_start - knowledge_time.astimezone(event_start.tzinfo)
 
 
+def x_years_ago_at_date(
+    event_start: datetime | pd.DatetimeIndex,
+    day: int,
+    month: int,
+    x: int = 1,
+    get_bounds: bool = False,
+    z: str | None = None,
+) -> timedelta | pd.TimedeltaIndex | tuple[timedelta, timedelta]:
+    """Compute the sensor's knowledge horizon to represent the event could be known since some date, `x` years ago.
+
+    For example, it can be used for a tax rate that changes annually and with a known publication date.
+
+    :param event_start:     Start of the event, used as an anchor for determining the knowledge horizon.
+    :param day:             Reference day of the month of the annual date to compare against.
+    :param month:           The month of the annual date to compare against.
+    :param x:               The number of years to shift the reference date to.
+    :param z:           Timezone string.
+    :param get_bounds:      If True, this function returns bounds on the possible return value.
+                            These bounds are normally useful for creating more efficient database queries when filtering by belief time.
+    """
+
+    MAX_DAYS_IN_A_YEAR = 366
+    MIN_DAYS_IN_A_YEAR = 364
+
+    if x <= 0:
+        raise ValueError("Only positive values for `x` are supported.")
+
+    if get_bounds:
+        # The minimum corresponds to an event at the 1st of January and a reference on the 31st of December on the the year `x` years ago.
+        # The maximum correponds to an event on the 31st of December and a reference on the 1st of January on the year `x` years ago.
```

Comment:
- corre**s**ponds
- ~on the 31st of December~ -> just before new year's midnight